runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/MonitoringUtil.java

```diff
@@ -67,8 +67,8 @@
   private static final String JOB_MESSAGE_DETAILED = "JOB_MESSAGE_DETAILED";
   private static final String JOB_MESSAGE_DEBUG = "JOB_MESSAGE_DEBUG";
 
-  private String projectId;
-  private Messages messagesClient;
+  private final String projectId;
+  private final DataflowClient dataflowClient;
```

Comment:
If the `projectId` was available from the `dataflowClient` then we could get rid of the duplication of these fields in multiple places.